### PR TITLE
match layout language to book language

### PR DESCRIPTION
**Pull request by bethshook**
for: openstax/unified#1552

### CODE REVIEW

File: src/app/MessageProvider.tsx
Comment (bethshook) [inline src/app/MessageProvider.tsx:23]: @TomWoodward should this now be called from within the component with the appropriate locale?

Comment (TomWoodward) [reply src/app/MessageProvider.tsx:23]: yea, and probably use state to wait until it finishes before you render

**Comment by TomWoodward** [comment]
you probably need to update the fixture book right?

**Comment by bethshook** [comment]
@TomWoodward it's in there

File: src/app/MessageProvider.tsx
Comment (TomWoodward) [inline src/app/MessageProvider.tsx:41]: its easier to use `userSelector` rather than connect now

File: src/gateways/createIntl.ts
Comment (TomWoodward) [inline src/gateways/createIntl.ts:7]: how much would it suck to make this async? it'd be nice to do a dynamic `import` rather than loading all the messages for every language

File: src/test/createTestServices.ts
Comment (TomWoodward) [inline src/test/createTestServices.ts:6]: gateways are supposed to be proxies to remote or third party services, i'd put `createIntl` in `messages`, probably move `MessageProvider` in there too

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/gateways/createIntl.ts
Comment (bethshook) [reply src/gateways/createIntl.ts:7]: so I have this working but the catch is now a whole lot of tests have to be made async (which is maybe what you were alluding to)

Comment (TomWoodward) [reply src/gateways/createIntl.ts:7]: yea i can see how a lot of tests would be broken... i'm struggling to think of a way to make this better without that problem.

i definitely don't want to be loading every language file and then picking one, and since we don't know the language until the book loads i'm not sure if there is any way around it

File: src/app/MessageProvider.tsx
Comment (TomWoodward) [inline src/app/MessageProvider.tsx:16]: this doesn't have to be defined in the function, you can `useEffect` with the locale as a dependency. something like:

```
useEffect(async () => {
  await polyfill(locale)
  setPolyfillLoaded(true);
}, [locale, polyfill])
```

File: src/app/MessageProvider.tsx
Comment (TomWoodward) [inline src/app/MessageProvider.tsx:33]: i thought you were gonna put something in the route or something?

File: src/app/MessageProvider.tsx
Comment (TomWoodward) [inline src/app/MessageProvider.tsx:40]: if the polyfill function was separate you wouldn't need the extra condition here

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/app/messages/MessageProvider.spec.tsx
Comment (bethshook) [inline src/app/messages/MessageProvider.spec.tsx:61]: @TomWoodward this is the test suite I'm updating. [here](https://github.com/openstax/rex-web/blob/master/src/app/MessageProvider.spec.tsx) you can see the original. I'm rendering the provider since `polyfill()` is now called from inside the component, but I think it's ignoring the polyfill module mocks.

**Review by PiotrKozlowski — DISMISSED** [review]
The code looks good, conflicts need to be resolved and I think there will be some lint errors due to comments before `runHooks` util function, since they are not longer required

File: src/app/content/highlights/components/utils/showConfirmation.tsx
Comment (bethshook) [inline src/app/content/highlights/components/utils/showConfirmation.tsx:6]: @TomWoodward I will need Bartek's changes after all

File: src/app/messages/createIntl.ts
Comment (bethshook) [inline src/app/messages/createIntl.ts:16]: @TomWoodward is this what you had in mind?

File: src/app/content/components/BookBanner.tsx
Comment (TomWoodward) [inline src/app/content/components/BookBanner.tsx:216]: this is gonna conflict with bartek's thing, leave it as services

File: src/app/content/components/Content.browserspec.tsx
Comment (TomWoodward) [inline src/app/content/components/Content.browserspec.tsx:13]: is this misplaced from the other pr?

File: src/app/content/highlights/components/utils/showConfirmation.tsx
Comment (TomWoodward) [inline src/app/content/highlights/components/utils/showConfirmation.tsx:9]: so i think you can actually just call createIntl and not even bother taking the intl argument, but bartek is gonna need dispatch so the services are gonna end up still in here

File: src/app/messages/createIntl.ts
Comment (TomWoodward) [inline src/app/messages/createIntl.ts:7]: i think i am with you that this doesn't have to be in services anymore. you can just export a function for `createIntl` and call it from anywhere, but its important that we not be re-loading the messages all the time, so you can check to see if the module system caches it, i can't remember if it does that automatically for inline imports. if it doesn't cache it by default you can wrap the whole definition in `lodash/fp/memoize`

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/app/content/components/Content.browserspec.tsx
Comment (bethshook) [reply src/app/content/components/Content.browserspec.tsx:13]: no, tests are failing mysteriously. I'll make the other changes and then deal with them

File: src/app/content/components/BookBanner.tsx
Comment (bethshook) [reply src/app/content/components/BookBanner.tsx:216]: does it need services other than intl? can I pass them in separately since we're talking about removing intl from services?

Comment (bethshook) [reply src/app/content/components/BookBanner.tsx:216]: nvm, saw your other comment

File: src/app/content/highlights/components/utils/showConfirmation.tsx
Comment (jarosik10) [inline src/app/content/highlights/components/utils/showConfirmation.tsx:13]: `ConfirmationModal` will always be in English?

**Review by jarosik10 — CHANGES_REQUESTED** [review]

File: src/app/types.ts
Comment (jarosik10) [inline src/app/types.ts:64]: I suppose it should be deleted.

**Review by jarosik10 — DISMISSED** [review]

File: src/app/content/highlights/components/utils/showConfirmation.tsx
Comment (bethshook) [reply src/app/content/highlights/components/utils/showConfirmation.tsx]: just a placeholder, sorry. will ping you once the `showConfirmation` logic is ready.

File: src/app/content/highlights/components/utils/showConfirmation.tsx
Comment (bethshook) [inline src/app/content/highlights/components/utils/showConfirmation.tsx:9]: @TomWoodward This is a placeholder as I'm confused about how to deal with typing given the change to `useServices`. Should I update create a new type that extends AppServices to include the redux middleware?

File: src/app/content/components/Page/PageComponent.tsx
Comment (TomWoodward) [inline src/app/content/components/Page/PageComponent.tsx:62]: this could be bad making this async, does it have to be async?

File: src/app/content/highlights/components/utils/showConfirmation.tsx
Comment (TomWoodward) [reply src/app/content/highlights/components/utils/showConfirmation.tsx:9]: you could just type this here as `AppServices & MiddlewareAPI`, pretty sure that is what the redux effect type does

File: src/app/content/utils/seoUtils.ts
Comment (TomWoodward) [inline src/app/content/utils/seoUtils.ts:114]: did you have to actually change these?

File: src/app/messages/MessageProvider.tsx
Comment (TomWoodward) [inline src/app/messages/MessageProvider.tsx:30]: technically the concept of books is out of scope this high up in the app, but i'm not sure i care enough to fix it. the right encapsulation would be to defer to the route and have the content route do something with the book, but i would rather get this merged than refine this

File: src/app/messages/createIntl.ts
Comment (TomWoodward) [inline src/app/messages/createIntl.ts:4]: you're importing the english messages unconditionally here, you need to defer this to the catch and if that also fails just explode everywhere

File: src/app/types.ts
Comment (TomWoodward) [inline src/app/types.ts:60]: this is the same as `AppServices & MiddlewareAPI`

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/app/messages/MessageProvider.tsx
Comment (bethshook) [reply src/app/messages/MessageProvider.tsx:30]: ok, I'll make a note to create a ticket once this is merged

File: src/app/messages/createIntl.ts
Comment (bethshook) [inline src/app/messages/createIntl.ts:20]: @TomWoodward Is this what you had in mind for the catch? If `enMessages` are not imported then the app won't render.

File: src/test/MessageProvider.tsx
Comment (bethshook) [inline src/test/MessageProvider.tsx:19]: @TomWoodward I removed the polyfill stuff from this test component since I use the real MessageProvider when testing that

File: src/app/content/hooks/receiveContent.ts
Comment (bethshook) [inline src/app/content/hooks/receiveContent.ts:40]: @TomWoodward this is what I mentioned re `services.intl.current` not being defined yet when this hook runs

**Review by TomWoodward — APPROVED** [review]

**Comment by Bkblodget** [comment]
@lenka222 the "symbol" issue you found above has been resolved in the latest version of content which can be viewed [here](https://rex-web.herokuapp.com/books/728df0bb-e07f-489d-91e3-4734a5932f92@1.122/pages/skorowidz-rzeczowy?archive=https%3A%2F%2Fopenstax.org%2Fapps%2Farchive-preview%2F20210804.143519).

The CE Styles team is also opening up an issue to update the recipe to display "Symbol" in Polish, just in case if in the future there does need to be a Symbols section in the index.

**Pull request merged**
